core/store/src/main/java/org/locationtech/geowave/core/store/api/DataStore.java

```diff
@@ -105,6 +107,27 @@
 	<T> CloseableIterator<T> query(
 			final Query<T> query );
 
+	/**
+	 * Returns all data in this data store that matches the query parameter and
+	 * gives the user the option of specifying if duplicates should be filtered
+	 * out or not. All data that matches the query will be returned as an
+	 * instance of the native data type. The Iterator must be closed when it is
+	 * no longer needed - this wraps the underlying scanner implementation and
+	 * closes underlying resources.
+	 *
+	 * @param query
+	 *            data constraints for the query and additional options for
+	 *            processing the query
+	 * @param filterDuplicates
+	 *            should the iterator filter out duplicates?
+	 * @return An iterator on all results that match the query. The iterator
+	 *         implements Closeable and it is best practice to close the
+	 *         iterator after it is no longer needed.
+	 */
+	<T> CloseableIterator<T> query(
```

Comment:
I am _really_ cautious about adding anything to `DataStore` because we are really trying hard to keep this interface as clean as possible.  In this case this method on the `DataStore` interface seems pretty unnecessary unless I'm missing something.  The `DuplicateDeletionCallback` can use `BaseDataStore` and it only needs access to `delete(Query, deleteDuplicates)` which in turn merely needs an extra flag on `internalQuery` and not the additional `query` method.  The only place I see the query used with the boolean for filterDuplicates is within the ITs which fundamentally I try not to make major code impacts to satisfy something in the ITs, to the extent possible.  In this case, it seems to me there's a couple of options available to the ITs to do the exact same thing - one is to just cast to BaseDataStore in the IT and assume these are all BaseDataStores that are being tested. Thats easiest but a bit more intrusive than what may be a little more work but completely unintrusive - within the IT you could also just overwrite the `DuplicateEntryCount` within the DataStatisticsStore to be 0 when you don't want to dedupe on query and 1 or whatever it was originally when you want it to behave normally on query/delete.  Its a hack, but it would be a hack in the IT that should get the job done without needing any extra code in the baseline.

Reply:
Totally agree, I added this originally for IT but thought it might be handy for users if they ever needed a sanity check on their data as well. I'll use a work around for the IT instead and remove.